### ISSUE TITLE
AnnotationRule formatting uncertainties

### ISSUE THREAD

**Issue by rom4ek** [pinterest/ktlint#556]
I have a following Retrofit interface:

```Kotlin
interface UserService {

    fun updateMarketingConsent(
        @Path("userId") userId: String,
        @Path("context") context: String,
        @Body body: MarketingConsent
    ): Completable

    fun showUser(@Path("id") userId: String, @Query("include") include: String? = null, @QueryMap fields: Map<String, String> = emptyMap()): Single

    fun logout(@Path("userId") userId: String): Completable
}
```

Running ktlint in `--format` mode gives the following output:

```Kotlin
interface UserService {

    fun updateMarketingConsent(
        @Path("userId")
        userId: String,
        @Path("context")
        context: String,
        @Body body: MarketingConsentStructure
    ): Completable

    fun showUser(
        @Path("id")
        userId: String,
        @Query("include") include: String? = null, @QueryMap fields: Map<String, String> = emptyMap()
    ): Single

    fun logout(
        @Path("userId")
        userId: String
    ): Completable
}
```

Imo, the latter is way less readable. I went through [the official styleguide](https://kotlinlang.org/docs/reference/coding-conventions.html#annotation-formatting), but I think an annotation with arguments should be allowed for method params (even though not sure that it's applicable for all possible cases).

Also in the `showUser` function, the last two parameters should be placed on different lines (so the with-argument annotation followed by no-argument annotation). Cause otherwise the `parameter-list-wrapping` rules kicks in. This is most likely a bug in formatter, right?

**Comment by shashachu** [comment]
Ah, yeah. I don't think the rule was intended for parameter annotations. We can put up a diff to have it ignore those annotations completely.

**Comment by francescocervone** [comment]
Is there a way to disable the annotation rule?

**Comment by rom4ek** [comment]
@francescocervone `disabled_rules=experimental:annotation` in your `.editorconfig`

**Comment by arturbosch** [comment]
+1 for this change

**Comment by Monchi** [comment]
I think readability and following official style are both important. I suggest to create two codestyle (https://github.com/pinterest/ktlint/issues/548), official one and strong one.
